kjdoc/tour.md

```diff
@@ -538,13 +538,441 @@ This section describes KJ APIs that control process execution and low-level inte
+    // Oh no, an error! Log it and end the loop.
+    KJ_LOG(ERROR, exception);
+    kj::throwFatalException(kj::mv(exception));
+  });
+}
+```
+
+The problem in this example is that the recursive call is _not_ a tail call, due to the `.catch_()` appended to the end. Every time around the loop, a new `.catch_()` is added to the promise chain. If an exception were thrown, that exception would end up being logged many times -- once for each time the loop has repeated so far.
+
+In this case, the best fix is to pull the `.catch_()` out of the loop entirely:
+
+```c++
+kj::Promise<void> boopEvery5Seconds(kj::Timer& timer) {
+  return boopEvery5SecondsLoop(timer)
+      .catch_([](kj::Exception&& exception) {
+    // Oh no, an error! Log it and end the loop.
+    KJ_LOG(ERROR, exception);
+    kj::throwFatalException(kj::mv(exception));
+  })
+}
+
+kj::Promise<void> boopEvery5SecondsLoop(kj::Timer& timer) {
+  // No memory leaks now!
+  timer.afterDelay(5 * kj::SECONDS).then([&timer]() {
+    KJ_LOG(WARNING, "boop");
+    // Loop by recursing.
+    return logEvery5Seconds(timer);
+  });
+}
+```
+
+Another possible fix would be to make sure the recursive continuation and then error handler are passed to the same `.then()` invocation:
+
+```c++
+kj::Promise<void> boopEvery5Seconds(kj::Timer& timer) {
+  // WARNING! MEMORY LEAK!
+  timer.afterDelay(5 * kj::SECONDS).then([&timer]() {
+    boop();
+  }).then([&timer]() {
+    // Loop by recursing.
+    return boopEvery5Seconds(timer);
+  }, [](kj::Exception&& exception) {
+    // Oh no, an error! Log it and end the loop.
+    KJ_LOG(ERROR, exception);
+    kj::throwFatalException(kj::mv(exception));
+  });
+}
+```
+
+Notice that in this second case, the error handler is scoped so that it does _not_ catch exceptions thrown by the recursive call; it only catches exceptions from `boop()`. This solves the problem, but it's a bit trickier to understand and to ensure that exceptions can't accidentally slip past the error handler.
+
+### Forking and splitting promises
+
+As mentioned above, `.then()` and similar functions consume the promise on which they are called, so can only be called once. But, what if you want to start multiple tasks using the result of a promise? You could solve this in a convoluted way using adapted promises, but KJ has a built-in solution: `.fork()`
+
+```c++
+kj::Promise<int> promise = ...;
+kj::ForkedPromise<int> forked = promise.fork();
+kj::Promise<int> branch1 = promise.addBranch();
+kj::Promise<int> branch2 = promise.addBranch();
+kj::Promise<int> branch3 = promise.addBranch();
+```
+
+A forked promise can have any number of "branches" which represent different consumers waiting for the same result.
+
+Forked promises use reference counting. The `ForkedPromise` itself, and each branch created from it, each represents a reference to the original promise. The original promise will only be canceled if all branches are canceled and the `ForkedPromise` itself is destroyed.
```

Comment:
Nit: grammar -- I think _The `ForkedPromise` itself, and each branch created from it, each represents ..._ has a plural subject, despite the coordinate making it plural also being a parenthetical expression ("and each branch created from it"). Honestly, though, it seems like it's a judgment call. See this: https://english.stackexchange.com/questions/56210/singular-or-plural-verb-form-where-subject-includes-a-parenthetical-element

My brain can't come up with a better way to word it; feel free to ignore.